main.go

```diff
@@ -177,17 +177,18 @@ func usage() {
 
 func locateDotGit(dir string) (string, error) {
 	absDir, err := filepath.Abs(dir)
+	rooted := ""
 	if err != nil {
 		return "", err
 	}
 
-	for absDir != "/" {
+	for absDir != rooted {
```

Comment:
So, instead of doing `absDir != "/"` we are doing `absDir != ""` and it does not break on Linux? How does it work?

Reply:
because we set rooted = to the previous absDir (191) before absDir is updated with filepath.Dir(absDir). If we are already at root filepath.Dir has no further traversing to do so it returns root again. This is our exit condition. rooted = root and absDir = root.

Reply:
@roman1900 ah, so `rooted` is some sort of `prevAbsDir`? I see, thanks! I'm gonna test on my machine a bit later and then merge it if everything's ok. Thanks for the contribution!